Userland/Libraries/LibJS/Runtime/Temporal/PlainDate.cpp

```diff
@@ -380,6 +381,38 @@ bool is_valid_iso_date(i32 year, u8 month, u8 day)
     return true;
 }
 
+// 3.5.6 DifferenceDate ( calendarRec, one, two, options ), https://tc39.es/proposal-temporal/#sec-temporal-differencedate
+ThrowCompletionOr<NonnullGCPtr<Duration>> difference_date(VM& vm, Object& calendar, PlainDate const& one, PlainDate const& two, Object& options)
+{
+    // FIXME: 1. Assert: one.[[Calendar]] and two.[[Calendar]] have been determined to be equivalent as with CalendarEquals.
+    // FIXME: 2. Assert: options is an ordinary Object.
+
+    // 3. Assert: options.[[Prototype]] is null.
+    VERIFY(!static_cast<Object const&>(options).prototype());
```

Comment:
This const cast looks like you spelled it weirdly :). But if the const method is public, you shouldn't need a cast at all.

Reply:
It's public, but since `Object* prototype() { return shape().prototype(); }` remains protected in `Object` and `options` is passed in as a `Object& options` (non-const reference). This is just to convince the compiler to call the const-qualified `prototype()` function, instead of the protected version.

hmm, one alternative may be to make both versions of this function public? Another might be to remove ` Object* prototype() { return shape().prototype(); }` and use `shape().prototype();` in `Object::internal_set_prototype_of`?

Reply:
Oh wait, hmm - actually, does this need to be non-const reference at all! I think I can also just make `CalendarDateUntil` take a `Object const&` and then it should hopefully remove the cast as well. But the above problem doesn't seem ideal in the future, so should perhaps have some type of change for that too :thonk: